src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -977,7 +978,15 @@ uint32 Unit::DealDamage(Unit* attacker, Unit* victim, uint32 damage, CleanDamage
 
         //if (attacker && victim->GetTypeId() == TYPEID_PLAYER && victim != attacker)
         //victim->ToPlayer()->UpdateAchievementCriteria(ACHIEVEMENT_CRITERIA_TYPE_TOTAL_DAMAGE_RECEIVED, health); // pussywizard: optimization
-        Unit::Kill(attacker, victim, durabilityLoss, cleanDamage ? cleanDamage->attackType : BASE_ATTACK, spellProto);
+
+        sAsyncCallbackMgr->AddAsyncCallback([attacker, victim, durabilityLoss, cleanDamage, spellProto]()
+        {
+            // Pussywarlock Elite MDic - Unit death not showing full spell animation.
+            // This delay of 400ms provides that opportunity for it to not be nerfed.
+            std::this_thread::sleep_for(400ms);
+
+            Unit::Kill(attacker, victim, durabilityLoss, cleanDamage ? cleanDamage->attackType : BASE_ATTACK, spellProto);
```

Comment:
what is guarantees that attacker and victim aren't deleted within the 400ms timeframe?